A test dependency is used in an exported code

### Bug description

In `omit-deep.d.ts` there is a new line that uses `expect-type` which is a devDependency.
```
import type {UnionToTuple} from 'expect-type';
```
This causes this TS validation error:
```
> tsc -p tsconfig.build.json

node_modules/type-fest/source/omit-deep.d.ts:1:33 - error TS2307: Cannot find module 'expect-type' or its corresponding type declarations.

1 import type {UnionToTuple} from 'expect-type';
                                  ~~~~~~~~~~~~~


Found 1 error in node_modules/type-fest/source/omit-deep.d.ts:1
```

### Repro

https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBDAnmApnA3geRMGARFFMAXzgDMoIQ4ByJVAWjJQGcYaBuAWAChf60ASQB2ZCAEY4AXgy8AkAEMAJkqisWALjgBteXPR65bNShhbjwYQHM9xADR6DPOS+OEYAJnMwolqw+cXMQhvX2tbALkAXW4eYn5kNABBFTUWFhExaThsXAIiAB5MiTtaZVV1ADpxSuCaAD5YoA

--- a/source/omit-deep.d.ts
+++ b/source/omit-deep.d.ts
@@ -1,4 +1,3 @@
-import type {UnionToTuple} from 'expect-type';
 import type {ArraySplice} from './array-splice';
 import type {ExactKey, IsArrayReadonly, NonRecursiveType, SetArrayAccess, ToString} from './internal';
 import type {IsEqual} from './is-equal';
@@ -6,6 +5,7 @@ import type {IsNever} from './is-never';
 import type {LiteralUnion} from './literal-union';
 import type {Paths} from './paths';
 import type {SimplifyDeep} from './simplify-deep';
+import type {UnionToTuple} from './union-to-tuple';
 import type {UnknownArray} from './unknown-array';
 
 /**

